apps/reference/docs/guides/functions.mdx

```diff
@@ -285,6 +285,10 @@ We recommend developing “fat functions”. This means that you should develop
 
 We recommend using hyphens to name functions because hyphens are the most URL-friendly of all the naming conventions (snake_case, camelCase, PascalCase).
 
+### CORS (Cross-Origin Resource Sharing)
+
+We recommend adding a check to handle [CORS Preflight](https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request) requests in your edge function. Otherwise you won't be able to invoke the function from browsers. Check this [code snippet for a example implemention](https://github.com/supabase/supabase/blob/master/examples/edge-functions/supabase/functions/browser-with-cors/index.ts#L11-L14).
```

Comment:
Since it's only a few lines, can we just add the example inline to save the user a click(not sure if the below would still be correct)? If not, let's pin to a commit (e.g., `https://github.com/supabase/supabase/blob/93702445373c1c5a18878752e734a4ac6ba24356/examples/edge-functions/supabase/functions/browser-with-cors/index.ts#L11-L14`) so the lines don't drift over time.

```suggestion
We recommend adding a check to handle [CORS Preflight](https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request) requests in your edge function to be able to invoke the function from browsers.

```ts
serve(async (req) => {
  if (req.method === 'OPTIONS') {
    return new Response('ok', { headers: corsHeaders })
  }
  ...
})
```
```

Reply:
yeah, that'd be good. Also, there's a link to the full example under Examples section.

Reply:
Hm, looks like something went wrong here:

![image](https://user-images.githubusercontent.com/5748289/192404254-0e61ccf3-1321-44db-a594-784d79c04575.png)

Also, could we still link to the example from here? Because the code snippet uses a variable called `corsHeaders` and folks might be confused as to what this is if they can't look into the code example. Alternatively we could define the `corsHeaders` inline in the example. Wdyt?

Reply:
ah shoot! Opened a PR to fix https://github.com/supabase/supabase/pull/9178